vector/src/main/java/im/vector/app/features/location/LocationSharingViewModel.kt

```diff
@@ -47,50 +65,106 @@ class LocationSharingViewModel @AssistedInject constructor(
     init {
         locationTracker.start(this)
         setUserItem()
-        createPin()
+        updatePin()
+        compareTargetAndUserLocation()
     }
 
     private fun setUserItem() {
         setState { copy(userItem = session.getUser(session.myUserId)?.toMatrixItem()) }
     }
 
-    private fun createPin() {
-        locationPinProvider.create(session.myUserId) {
-            setState {
-                copy(
-                        pinDrawable = it
-                )
+    private fun updatePin(isUserPin: Boolean? = true) {
+        if (isUserPin == true) {
```

Comment:
We generally prefer `if (isUserPin.orFalse())`.

Reply:
Thanks, I was not aware of this extension.